controllers/storagecluster/reconcile.go

```diff
@@ -513,7 +520,7 @@ func (obj *ocsCephConfig) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.S
 		}
 	}
 	val, ok := found.Data["config"]
-	if !ok || val != rookConfigData || !ownerRefFound {
+	if !ok || (val != defaultRookConfigData && val != sc.Spec.RookConfig) || !ownerRefFound {
```

Comment:
I think we should compare it against `configData`. This current comparison would always lead to updating the config map if `RookConfig`    is not set in the storagecluster spec.

Reply:
And if we named the variable `rookConfigData` above (instead of `configData`,  then this code would remain unchanged! 😁